cmd/pulumi-test-language/interface.go

```diff
@@ -1071,8 +1073,36 @@ func (eng *languageTestServer) RunLanguageTest(
 			SecretsManager:     sm,
 			SecretsProvider:    b64secrets.Base64SecretsProvider,
 			Scopes:             backend.CancellationScopes,
+		}
+
+		assertPreview := run.AssertPreview
+		if assertPreview == nil {
+			// if no assertPreview is provided for the test run, we creat a default implementation
+			// where we simply assert that the preview changes did not error
+			assertPreview = func(l *tests.L, proj string, err error, p *deploy.Plan, changes display.ResourceChanges) {
+				assert.NoErrorf(l, err, "expected no error in preview")
+			}
+		}
+
+		// Perform a preview on the stack
+		plan, previewChanges, res := s.Preview(ctx, updateOperation, nil)
```

Comment:
I guess this is going to impact the runtime of our tests -- do we know by how much? I'm very pro this change but e.g. if this is significant we could tweak this so that e.g. if `assertPreview` is nil, we _don't_ run a preview. That might be a terrible idea though.

Reply:
> I guess this is going to impact the runtime of our tests -- do we know by how much? 

Not really, I was surprised I didn't need to increase the timeout of the tests so was happy with that

> if assertPreview is nil, we don't run a preview. That might be a terrible idea though.

I think running preview for every test and checking that it didn't fail is a bare minimum to get the conformance tests on par with our current integration tests so that we can eventually remove the existing integration tests. Many of them already overlap with conformance tests so we need this to start trimming down the integration ones

Reply:
Works for me -- if we think the runtime change is negligible/worth it, let's do it!